lumicks/pylake/kymotracker/detail/tracking.py

```diff
@@ -0,0 +1,180 @@
+        Calibrated kymograph channel.
+    line_width : float
+        Expected line width in pixels.
+    pixel_threshold : float
+        Intensity threshold for the pixels. Local maxima above this intensity level will be
+        designated as a line origin.
+    window : int
+        Number of kymograph lines in which the particle is allowed to disappear (and still be part
+        of the same line).
+    sigma : float or None
+        Uncertainty in the particle position. This parameter will determine whether a peak in the
+        next frame will be linked to this one. Increasing this value will make the algorithm tend
+        to allow more positional variation in the lines. If none, the algorithm will use half the
+        line width.
+    vel : float
+        Expected velocity of the traces in the image. This can be used for non-static particles that
+        are expected to move at an expected rate (default: 0.0).
+    diffusion : float
+        Expected diffusion constant (default: 0.0). This parameter will influence whether a peak in
+        the next frame will be connected to this one. Increasing this value will make the algorithm
+        allow more positional variation in.
+    sigma_cutoff : float
+        Sets at how many standard deviations from the expected trajectory a particle no longer
+        belongs to this trace. Lower values result in traces being more stringent in terms of
+        continuing (default: 2.0).
+    rect : tuple of two pixel coordinates
+        Only perform tracking over a subset of the image. Pixel coordinates should be given as:
+        ((min_time, min_coord), (max_time, max_coord)).
+
+    References
+    ----------
+    [1] Sbalzarini, I. F., & Koumoutsakos, P. (2005). Feature point tracking and trajectory analysis
+    for video imaging in cell biology. Journal of structural biology, 151(2), 182-195.
+    [2] Mangeol, P., Prevo, B., & Peterman, E. J. (2016). KymographClear and KymographDirect: two
+    tools for the automated quantitative analysis of molecular and cellular dynamics using
+    kymographs. Molecular biology of the cell, 27(12), 1948-1957.
+    """
+    data_selection = kymograph_channel.get_rect(rect) if rect else kymograph_channel.data
+
+    sigma = sigma if sigma else 0.5 * line_width
+    coordinates, time_points = peak_estimate(
+        data_selection, np.ceil(0.5 * line_width), pixel_threshold
+    )
+    if len(coordinates) == 0:
+        return []
+
+    peaks = KymoPeaks(
+        *refine_peak_based_on_moment(
+            data_selection, coordinates, time_points, np.ceil(0.5 * line_width)
+        )
+    )
+    peaks = merge_close_peaks(peaks, np.ceil(0.5 * line_width))
+    lines = points_to_line_segments(
+        peaks,
+        kymo_score(vel=vel, sigma=sigma, diffusion=diffusion),
+        window=window,
+        sigma_cutoff=sigma_cutoff,
+    )
+
+    lines = [KymoLine(line.time_idx, line.coordinate_idx, kymograph_channel) for line in lines]
+
```

Comment:
would it be worthwhile to have a constructor classmethod that specifically takes `(KymoLineData, CalibratedKymographChannel)` as inputs? On one hand, this clearly couples `KymoLine` and `KymoLineData`, but that might not be a bad thing. Clearly the two classes are related just based on the names. This then defines a very specific interface for converting from one to the other. 

On the other hand it could just end up being unneeded syntactic sugar